src/main/java/org/opensearch/neuralsearch/search/HybridTopFieldDocSortCollector.java

```diff
@@ -0,0 +1,485 @@
+/*
```

Comment:
At high level , this class says Abstract but it contains 500 lines of code , it's very hard to maintain and read from one file, Can u break this class in logical part.

Reply:
+1. Seems like this PR is favoring inheritance for code reuse. Can we keep inheritance/polymorphism for enforcing behavior and composition for code reuse? This has added a advantages and you will find quite a few articles around it. Here is a pointer to start with https://en.wikipedia.org/wiki/Composition_over_inheritance 

As a guideline don't use inheritance for code reuse until absolutely required